versioned_hdf5/slicetools.pyx

```diff
@@ -234,6 +238,125 @@ cpdef build_data_dict(dcpl, raw_data_name: str):
     return data_dict
 
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.infer_types(True)
+cpdef tuple[np.ndarray, np.ndarray] build_slab_indices_and_offsets(
+    dcpl,
+    shape: tuple[int, ...],
+    chunk_size: tuple[int, ...],
+):
+    """Build the slab_indices and slab_offsets arrays for a StagedChangesArray that
+    accesses a versioned virtual dataset.
```

Comment:
These are explained in #386 -> staged_changes.py -> docstring of StagedChangesArray